test/e2e_node/pod_resize_test.go

```diff
@@ -307,56 +309,161 @@ func verifyPodContainersCgroupValues(ctx context.Context, f *framework.Framework
-			var expectedCPUShares int64
-			var expectedCPULimitString, expectedMemLimitString string
-			expectedMemLimitInBytes := tc.Resources.Limits.Memory().Value()
-			cpuRequest := tc.Resources.Requests.Cpu()
-			cpuLimit := tc.Resources.Limits.Cpu()
-			if cpuRequest.IsZero() && !cpuLimit.IsZero() {
-				expectedCPUShares = int64(kubecm.MilliCPUToShares(cpuLimit.MilliValue()))
-			} else {
-				expectedCPUShares = int64(kubecm.MilliCPUToShares(cpuRequest.MilliValue()))
-			}
-			cpuQuota := kubecm.MilliCPUToQuota(cpuLimit.MilliValue(), kubecm.QuotaPeriod)
-			if cpuLimit.IsZero() {
-				cpuQuota = -1
-			}
-			expectedCPULimitString = strconv.FormatInt(cpuQuota, 10)
-			expectedMemLimitString = strconv.FormatInt(expectedMemLimitInBytes, 10)
-			if podOnCgroupv2Node {
-				if expectedCPULimitString == "-1" {
-					expectedCPULimitString = "max"
+		var cpuRequest, cpuLimit *resource.Quantity
+		var expectedMemLimitInBytes int64
+		if ci.Resources != nil {
+			tc, _ := makeTestContainer(ci)
+			if tc.Resources.Limits != nil || tc.Resources.Requests != nil {
+				var expectedCPUShares int64
+				var expectedCPULimitString, expectedMemLimitString string
+				expectedMemLimitInBytes = tc.Resources.Limits.Memory().Value()
+				cpuRequest = tc.Resources.Requests.Cpu()
+				cpuLimit = tc.Resources.Limits.Cpu()
+				if cpuRequest.IsZero() && !cpuLimit.IsZero() {
+					expectedCPUShares = int64(kubecm.MilliCPUToShares(cpuLimit.MilliValue()))
+				} else {
+					expectedCPUShares = int64(kubecm.MilliCPUToShares(cpuRequest.MilliValue()))
 				}
-				expectedCPULimitString = fmt.Sprintf("%s %s", expectedCPULimitString, CPUPeriod)
-				if expectedMemLimitString == "0" {
-					expectedMemLimitString = "max"
+				cpuQuota := kubecm.MilliCPUToQuota(cpuLimit.MilliValue(), kubecm.QuotaPeriod)
+				if cpuLimit.IsZero() {
+					cpuQuota = -1
 				}
-				// convert cgroup v1 cpu.shares value to cgroup v2 cpu.weight value
-				// https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2254-cgroup-v2#phase-1-convert-from-cgroups-v1-settings-to-v2
-				expectedCPUShares = int64(1 + ((expectedCPUShares-2)*9999)/262142)
-			}
-			if expectedMemLimitString != "0" {
-				err := verifyCgroupValue(ci.Name, cgroupMemLimit, expectedMemLimitString)
+				expectedCPULimitString = strconv.FormatInt(cpuQuota, 10)
+				expectedMemLimitString = strconv.FormatInt(expectedMemLimitInBytes, 10)
+				if podOnCgroupv2Node {
+					if expectedCPULimitString == "-1" {
+						expectedCPULimitString = "max"
+					}
+					expectedCPULimitString = fmt.Sprintf("%s %s", expectedCPULimitString, CPUPeriod)
+					if expectedMemLimitString == "0" {
+						expectedMemLimitString = "max"
+					}
+					// convert cgroup v1 cpu.shares value to cgroup v2 cpu.weight value
+					// https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2254-cgroup-v2#phase-1-convert-from-cgroups-v1-settings-to-v2
+					expectedCPUShares = int64(1 + ((expectedCPUShares-2)*9999)/262142)
```

Comment:
nit: redundant type conversion
```suggestion
					expectedCPUShares = 1 + ((expectedCPUShares-2)*9999)/262142
```

Reply:
Done. Thanks.